builtin/providers/aws/resource_aws_ssm_association.go

```diff
@@ -0,0 +1,121 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSsmAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSsmAssociationCreate,
+		Read:   resourceAwsSsmAssociationRead,
+		Delete: resourceAwsSsmAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"parameters": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsSsmAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	log.Printf("[DEBUG] SSM association create: %s", d.Id())
+
+	assosciationInput := &ssm.CreateAssociationInput{
+		Name:       aws.String(d.Get("name").(string)),
+		InstanceId: aws.String(d.Get("instance_id").(string)),
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		assosciationInput.Parameters = expandDocumentParameters(v.(map[string]interface{}))
+	}
+
+	resp, err := ssmconn.CreateAssociation(assosciationInput)
+
+	if err != nil {
+		l, _ := ssmconn.ListDocuments(nil)
+
+		log.Printf("[DEBUG] documents found %s", l.DocumentIdentifiers)
+
+		return errwrap.Wrapf("[ERROR] Error creating SSM association: {{err}}", err)
+	}
+
+	d.SetId(*resp.AssociationDescription.Name)
+
+	return resourceAwsSsmAssociationRead(d, meta)
+}
+
+func resourceAwsSsmAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	log.Printf("[DEBUG] Reading SSM Assosciation: %s", d.Id())
+
+	params := &ssm.DescribeAssociationInput{
+		Name:       aws.String(d.Get("name").(string)),
+		InstanceId: aws.String(d.Get("instance_id").(string)),
+	}
+
+	resp, err := ssmconn.DescribeAssociation(params)
+
+	if err != nil {
```

Comment:
Is there a chance that someone will manually delete the association from the console? What will happen if so?